api/src/adapters/search/opensearch_query_builder.py

```diff
@@ -177,12 +177,23 @@ def filter_terms(self, field: str, terms: list) -> typing.Self:
         self.filters.append({"terms": {field: terms}})
         return self
 
+    def filter_should_prefix(self, field: str, value: str) -> typing.Self:
+        """
+        Adds a prefix filter under a should clause for the specified field.
+
+        This is used to match documents where the field value starts with the given prefix.
+        The filter is added to a `bool.should` context, allowing OR-style matching when combined
+        with other should clauses.
+        """
+        self.should.append({"prefix": {field: f"{value}-"}})
```

Comment:
Don't put the `-` in the builder, if we wanted to have prefix filters for things that aren't agencies they wouldn't work unless they had the exact same behavior.